docs: fix instructions for setting up `ProductDetailsComponent` in `start-routing.md`

gkalpak opened the pull request:
PR #34934 switched the `getting-started` docs example from using the index of a product in the `products` array to using the product's ID property for indentifiying each product in the `ProductDetailsComponent` component. However, some necessary changes in the example code and the `start-routing.md` guide were missed in #34934, resulting in broken instructions for the readers (see #40189).

This PR is essentially a follow-up to #34934, making the remaining changes in the example code and the guide instructions.

Fixes #40189.

##
This PR also includes some refactorings and clean-ups for the `getting-started`/`getting-started-v0` docs examples. See individual commits for more details.

mary-poppins commented:
You can preview bec84d8 at https://pr40197-bec84d8.ngbuilds.io/.

aikithoughts reviewed (approved):
LGTM, but I think this change overlaps with https://github.com/angular/angular/pull/40284.

Reviewed-for: global-docs-approvers

zytek commented:
The tutorial on angular.io uses product.id but sample StackBlitz project lacks `id` fields in products.ts

gkalpak committed:
docs: more closely align the `getting-started-v0` and `getting-started` docs examples

This commit aligns the code and style in the `getting-started-v0` docs
example with that of the `getting-started` example more closely (except
for the the parts that are intentionally different between the two).

gkalpak committed:
docs: clean up docregions in the `getting-started` docs example

This commit does some minor clean-up in docregions of the
`getting-started` docs example. More specifically, it:
- Removes redundant docregion markers.
- Fixes whitespace inconsistencies.
- Adds `/* ... */` to indicate that there is more code before the end of
  a docregion (in order to avoid confusion).

gkalpak committed:
docs: fix instructions for setting up `ProductDetailsComponent` in `start-routing.md`

PR #34934 switched the `getting-started` docs example from using the
index of a product in the `products` array to using the product's ID
property for indentifiying each product in the `ProductDetailsComponent`
component. However, some necessary changes in the example code and the
`start-routing.md` guide were missed in #34934, resulting in broken
instructions for the readers (see #40189).

This commit is essentially a follow-up to #34934, making the remaining
changes in the example code and the guide instructions.

Fixes #40189

gkalpak committed:
docs: refactor code in `get-product` docregion of `getting-started` docs example

This commit slightly refactors the code in the `get-product` docregion
of the `getting-started` docs example to make it easier to follow.

mary-poppins commented:
You can preview 0e15815 at https://pr40197-0e15815.ngbuilds.io/.
